### DISCUSSION TITLE
Reference implementation

### DISCUSSION THREAD

**Discussion by waghanza** [the-benchmarker/web-frameworks#3789] [Ideas]
Hi @leocavalcante,

Following https://github.com/the-benchmarker/web-frameworks/pull/3786#issuecomment-761560767, your idea seems awesome.

However this looks like like a subproject 😛  (and specially designed for `php` / `swoole`).

Regards,

/cc @sy-records 

PS : Discussions seems more made for this kind of ideas :heart:

**Comment by leocavalcante** [comment]
I think other languages could have some *references* too, I mean, for `javascript` there could be a [vanilla `node.js`](https://nodejs.org/docs/latest-v13.x/api/http.html) implementation; for `dart` there could be an implementation using only [`dart:io` package](https://dart.dev/tutorials/server/httpserver); `go` using only the [`net/http`](https://golang.org/pkg/net/http/); and so on.

PHP itself could have a:
- `vanilla` (with something even simpler than `basicphp`).
- `swoole` (pure Swoole implementation).
  - Also, other players can join the playground (`reactphp`, `amp`, `workerman`...)

**Reply by leocavalcante** [reply]
Actually, just crossed my mind that [a pure `workerman` version is already there](https://github.com/the-benchmarker/web-frameworks/blob/master/php/workerman/start.php). It is not labeled as a *reference* for other frameworks with this flavor, but...

**Comment by waghanza** [comment]
Sure, this only caveats I notice is that this task require a lot of dev.

I prefer to merge this only if we can have the same for all language, I mean on reference implementation by language (not easy but doable)

wdyt @the-benchmarker/web-frameworks ? seems a good idea

**Comment by ahopkins** [comment]
Speaking about Python... it certainly is possible and not overly complicated to build a small server to respond to endpoints using very low level blocks. Trying to determine _the_ single way to do it to be a base performance indicator? That is complicated. Happy to help though.
